src/js/libcs/Essentials.js

```diff
@@ -3,82 +3,98 @@
-operators['~>'] = 300; //approx greater
-operators['=:='] = 300; //Equals after evaluation
-operators['>='] = 300; //Größergleich
-operators['<='] = 300; //Kleinergleich
-operators['~>='] = 300; //ungefähr Größergleich
-operators['~<='] = 300; //ungefähr Kleinergleich
-operators['>'] = 300; //Größer
-operators['<'] = 300; //Kleiner
-operators['<>'] = 300; //Ungleich
-operators['&'] = 350; //Logisches Und
-operators['%'] = 350; //Logisches Oder
-operators['!='] = 350; //Ungleich
-operators['~!='] = 350; //ungefähr Ungleich
-operators['..'] = 350; //Aufzählung 1..5=(1,2,3,4,5)
-operators['++'] = 370; //Listen Aneinanderhängen
-operators['--'] = 370; //Listen wegnehmen
-operators['~~'] = 370; //Gemeinsame Elemente
-operators[':>'] = 370; //Append List
-operators['<:'] = 370; //Prepend List
-operators['='] = 400; //Zuweisung
-operators[':='] = 400; //Definition
-operators[':=_'] = 400; //Definition
-operators['::='] = 400; //Definition
-operators['->'] = 400; //Modifier
-operators[','] = 500; //Listen und Vektoren Separator
-operators[';'] = 500; //Befehlsseparator
-
+operators[':'] = 20; // Access to user-defined fields
+operators['.'] = 25; // Field access
+operators['°'] = 25; // Degree
+operators['_'] = 50; // x_i means i-th element of x
+operators['^'] = 50; // power
+operators['*'] = 100; // multiplication (also for vectors and matrices)
+operators['/'] = 100; // division (also vectoren divided by scalar)
+operators['+'] = 200; // addition (also for vectors and matrices)
+operators['-'] = 200; // subtraction (also for vectors and matrices)
+operators['!'] = 200; // logical not (unary)
+operators['=='] = 300; // equal
+operators['~='] = 300; // approximately equal
+operators['~<'] = 300; // not greater or approximately equal
+operators['~>'] = 300; // not less or approximately equal
+operators['=:='] = 300; // equals after evaluation
+operators['>='] = 300; // greater or equal
+operators['<='] = 300; // less or equal
+operators['~>='] = 300; // greater or approximately equal
+operators['~<='] = 300; // less or approximately equal
+operators['>'] = 300; // greater
+operators['<'] = 300; // less
+operators['<>'] = 300; // not equal
+operators['&'] = 350; // logical and
+operators['%'] = 350; // logical or
+operators['!='] = 350; // not equal
+operators['~!='] = 350; // not approximately equal
+operators['..'] = 350; // sequence
+operators['++'] = 370; // concatenation of lists, union of shapes
+operators['--'] = 370; // list or shape difference
+operators['~~'] = 370; // list or shape intersection
+operators[':>'] = 370; // append element to list
+operators['<:'] = 370; // prepend element to list
+operators['='] = 400; // assignment
+operators[':='] = 400; // function definition
+operators[':=_'] = 400; // function undefinition
+operators['::='] = 400; // binding function definition
+operators['->'] = 400; // modifier
+operators[';'] = 500; // sequence of statements
+//operators[','] = 510; // list element separator, handled in parselist
```

Comment:
At first, I wondered if this was deliberately commented out.

I would turn this into a mere comment, e.g.
```javascript
// The list element separator ',' (operator precedence 510) is handled in parselist.
```